src/doc/src/faq.md

```diff
@@ -259,3 +259,43 @@ Some issues we've seen historically which can cause crates to get rebuilt are:
 If after trying to debug your issue, however, you're still running into problems
 then feel free to [open an
 issue](https://github.com/rust-lang/cargo/issues/new)!
+
+### Why does Cargo.lock conflict?
+
+`Cargo.lock` contains exact information about your dependencies. It should be 
+generated and maintained by Cargo and should not be edited manually. If there 
+are non-trivial conflicts, you can reset all "yours" changes, fix all other 
+conflicts in the branch, and then run some cargo command (like `cargo tree` or 
+`cargo check`), which should re-update the lockfile with your own local changes. 
+If you previously ran some `cargo update` commands in your branch, you can re-run 
+them that this time.
+
+Besides that, there are other situations that create conflicts. Unfortunately, in 
+many cases these conflicts cannot be resolved automatically and need to be resolved 
+manually:
+
+- The project and its dependencies use [links][links] to repeatedly link the local 
+library.Cargo forbids linking two packages with the same native library, so even with 
+multiple layers of dependencies it is not allowed. At this time, you need to 
+manually delete the redundant link value. however, that there are [conventions in 
+place][conventions-in-place] to alleviate this.
+
+- When depending on different crates in the project, if these crates use the same 
+dependent library, but the version used is restricted, making it impossible to 
+determine the correct version, it will also cause conflicts.
+
+- If there are multiple versions of dependencies in the project, when using 
+[`direct-minimal-versions`][direct-minimal-versions], the minimum version requirements 
+cannot be met, which will cause conflicts.
+
+- If the dependent crate does not have the features you choose, it will also cause 
+conflicts. At this time, you need to check the dependent version and its features;
```

Comment:
These look valid and well-written from the angle of “versions conflict” at runtime. However, it doesn't really directly contribute to `Cargo.lock` file conflicts. I think we may change the heading to something like `What does “version's conflict” mean and how to resolve it?`

And maybe this FAQ should be put in this way:

> > failed to select a version for `x` which could resolve this conflict
>
> Have you seen the error message above?
> 
> This is one of the most annoying error message for Cargo users. There are several situations may lead us to a version conflict. Below we'll walk through possible causes and provide diagnostic techniques to help you out there:
>
> (list possible situations, steps to diagnose, and ways to work around respectively…)
>